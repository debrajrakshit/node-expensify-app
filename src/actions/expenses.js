import uuid from 'uuid';

// ADD_EXPENSE
// desctructure the first argument
// if doesn't exist destructure an empty object - lecture 93 (section 10)
// define all of the stuffs want to grab -  define as empty string or as per object
export const addExpense = (
    { 
        description = '',
        note = '',
        amount= 0,
        createdAt = 0 
    } = {}
) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amount,
        createdAt
    }
});

// REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
    type: 'REMOVE_EXPENSE',
    id
});

// EDIT_EXPENSE
export const editExpense = (id, updates) => ({
    type: 'EDIT_EXPENSE',
    id,
    updates
});

