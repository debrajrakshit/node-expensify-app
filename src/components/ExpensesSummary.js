import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';


// class ExpensesSummary extends React.Component {
//     state = {
//         expenses: selectExpenses()
//     };

//     render() {
//         return 
//     };
// };

export const ExpensesSummary = (props) => {
    const expensesCount = props.expenses.length;
    const expensesTotal = numeral((selectExpensesTotal(props.expenses)) / 100).format('$0,0.00');
    
    return (
        <div>
            <p>Viewing <span className='total-num'>{expensesCount}</span> {expensesCount > 1 ? 'expenses' : 'expense'} totalling {expensesTotal}</p>
        </div>
    )
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);