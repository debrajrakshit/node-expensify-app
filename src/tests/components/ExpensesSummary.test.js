import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render ExpensesSummary properly with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);

    expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary properly with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    
    expect(wrapper).toMatchSnapshot();
});