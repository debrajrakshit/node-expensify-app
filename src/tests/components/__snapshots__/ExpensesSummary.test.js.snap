// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary properly with 1 expense 1`] = `
<div>
  <p>
    Viewing 
    <span
      className="total-num"
    >
      1
    </span>
     
    expense
     totalling 
    $1.96
  </p>
</div>
`;

exports[`should render ExpensesSummary properly with multiple expense 1`] = `
<div>
  <p>
    Viewing 
    <span
      className="total-num"
    >
      3
    </span>
     
    expenses
     totalling 
    $1,141.96
  </p>
</div>
`;
